script/remote_db_tunnel.sh

```diff
@@ -0,0 +1,215 @@
+[ -z "${PORT}" ] && PORT=5432
+[ -z "${RDS_SECRET_NAME}" ] && RDS_SECRET_NAME='rds-postgresql-instance-output' #most used name for the rds instance secret.
+[ -z "${IMAGE}" ] && IMAGE='ministryofjustice/port-forward'
+
+if [ -z "${PORT_FORWARD_CONTAINER_NAME}" ]
+then
+  if [ -z "${CAS_PORT_FORWARD_CONTAINER_NAME}" ]
+  then
+    PORT_FORWARD_CONTAINER_NAME='port-forward-pod'
+  else
+    PORT_FORWARD_CONTAINER_NAME=${CAS_PORT_FORWARD_CONTAINER_NAME}
+  fi
+fi
+
+echo "---------------------------------------------------------------"
+echo "The following parameters will be used:"
+echo ""
+echo "* Namespace: $NAMESPACE"
+echo "* Port: $PORT"
+echo "* RDS Secret Name: $RDS_SECRET_NAME"
+echo "* Port Forward Container Name: $PORT_FORWARD_CONTAINER_NAME"
+echo "* Port Forward Image: $IMAGE"
+echo ""
+
+# ensure we're in the right k8s context
+kubectl config use-context live.cloud-platform.service.justice.gov.uk
+
+secrets=$(kubectl get secrets $RDS_SECRET_NAME --namespace "$NAMESPACE" -o json | jq ".data | map_values(@base64d)")
+rds_instance_endpoint=$(echo "$secrets" | jq -r '.rds_instance_address')
+database_name=$(echo "$secrets" | jq -r '.database_name')
+database_username=$(echo "$secrets" | jq -r '.database_username')
+database_password=$(echo "$secrets" | jq -r '.database_password')
+
+echo "---------------------------------------------------------------"
+echo "Checking if port forward container already exists"
+echo ""
+kubectl get pod $PORT_FORWARD_CONTAINER_NAME -n "$NAMESPACE"
+POD_EXISTS_RETURN_CODE=$?
+echo ""
+
+if [ $POD_EXISTS_RETURN_CODE -gt 0 ]
+then
+    echo "Creating pod..."
+    kubectl -n "$NAMESPACE" run $PORT_FORWARD_CONTAINER_NAME --image=$IMAGE \
+        --port=$PORT \
+        --env="REMOTE_HOST=$rds_instance_endpoint" \
+        --env="LOCAL_PORT=$PORT" \
+        --env="REMOTE_PORT=$PORT"
+    kubectl wait --for=condition=ready pod/$PORT_FORWARD_CONTAINER_NAME -n "$NAMESPACE"
+    echo "Port-forward pod $PORT_FORWARD_CONTAINER_NAME created!"
+else
+    echo "Port-forward pod already exists"
+fi
+
+if [ "$DISPLAY_CREDS" -eq 1 ]
+then
+    echo ""
+    echo "---------------------------------------------------------------"
+    echo "Connect using the following details in your local Database IDE"
+    echo ""
+    echo "* Host: localhost"
+    echo "* Database Name: $database_name"
+    echo "* User: $database_username"
+    echo "* Password: $database_password"
+    echo "* Port: $PORT"
+    echo ""
+fi
+
+if [ $CLI -eq 0 ]
+then
+    echo "Forwarding port. Use ctl-c to close port forwarding"
+    kubectl -n "$NAMESPACE" port-forward $PORT_FORWARD_CONTAINER_NAME $PORT:$PORT
+else
+    # This is ran in the background so we can start the cli
+    # we use trap at the top of this script to kill it on exit
+    kubectl -n "$NAMESPACE" port-forward $PORT_FORWARD_CONTAINER_NAME $PORT:$PORT &
+    port_forward_pid=$!
+    echo "Port forward pid is $port_forward_pid"
+
+    sleep 1
```

Comment:
QQ - what's the purpose of the sleep?

Reply:
Good question - i think it was returning before the container was ready (presumably by milliseconds) but let me try removing it and seeing if it's stable as it shouldn't really be required. If not stable i'll look at adding a check to make sure the containers ready instead.

Reply:
i've removed the sleep and it seems to work without issue. you already have a 'pod ready' check earlier on in the script, so i'm not sure why i added this!